go/vt/discovery/healthcheck.go

```diff
@@ -135,18 +144,26 @@ const (
 // ParseTabletURLTemplateFromFlag loads or reloads the URL template.
 func ParseTabletURLTemplateFromFlag() {
 	tabletURLTemplate = template.New("")
-	_, err := tabletURLTemplate.Parse(*TabletURLTemplateString)
+	_, err := tabletURLTemplate.Parse(TabletURLTemplateString)
 	if err != nil {
 		log.Exitf("error parsing template: %v", err)
 	}
 }
 
 func init() {
-	// Flags are not parsed at this point and the default value of the flag (just the hostname) will be used.
-	ParseTabletURLTemplateFromFlag()
-	flag.Var(&tabletFilters, "tablet_filters", "Specifies a comma-separated list of 'keyspace|shard_name or keyrange' values to filter the tablets to watch")
+	servenv.OnParseFor("vtgate", registerDiscoveryFlags)
+	servenv.OnParseFor("vtcombo", registerDiscoveryFlags)
+}
+
+func registerDiscoveryFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&TabletURLTemplateString, "tablet_url_template", "http://{{.GetTabletHostPort}}", "format string describing debug tablet url formatting. See the Go code for getTabletDebugURL() how to customize this.")
+	fs.DurationVar(&refreshInterval, "tablet_refresh_interval", 1*time.Minute, "tablet refresh interval")
+	fs.BoolVar(&refreshKnownTablets, "tablet_refresh_known_tablets", true, "tablet refresh reloads the tablet address/port map from topo in case it changes")
+	fs.IntVar(&topoReadConcurrency, "topo_read_concurrency", 32, "concurrent topo reads")
+	fs.StringSliceVar(&tabletFilters, "tablet_filters", []string{}, "Specifies a comma-separated list of 'keyspace|shard_name or keyrange' values to filter the tablets to watch")
 	topoproto.TabletTypeListVar(&AllowedTabletTypes, "allowed_tablet_types", "Specifies the tablet types this vtgate is allowed to route queries to")
```

Comment:
The issue is that function uses the global flagset (`pflag.Var`) instead of taking one as an argument (eg the `fs` parameter here). We need a second function for now, and then can delete the old one when we're done migrating.

Reply:
This is the only usage of that function so we can just change it to pass in the FlagSet. The funny thing is that there are at least 2 other flags that are comma-separated lists of tablet types, and those are defined as `string` and we parse the string after reading it.